### ISSUE TITLE
a crash for 2.10.0

### ISSUE THREAD

**Issue by minjie0707** [stripe/stripe-terminal-android#234]
<img width="1071" alt="image" src="https://user-images.githubusercontent.com/82426849/172177651-26f459b9-a4a2-4892-beb7-789caefc851a.png">
This crash is only happened on  2.10.0 , is fine on 2.9.0

**Comment by anton8901** [comment]
I have the same issue using the example app

```
2022-06-07 23:51:27.354 7296-7296/com.stripe.example.javaapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.stripe.example.javaapp, PID: 7296
    java.lang.NoClassDefFoundError: Failed resolution of: Ljava/time/Duration;
        at com.stripe.core.logging.dagger.MetricLoggerModuleKt.<clinit>(MetricLoggerModule.kt:26)
        at com.stripe.core.logging.dagger.MetricLoggerModule.<init>(MetricLoggerModule.kt:31)
        at com.stripe.core.logging.dagger.MetricLoggerModule.<init>(MetricLoggerModule.kt)
        at com.stripe.stripeterminal.DaggerTerminalComponent$Builder.build(DaggerTerminalComponent.java:1679)
        at com.stripe.stripeterminal.Terminal$Companion.initTerminal(Terminal.kt:860)
        at com.stripe.stripeterminal.Terminal$Companion.initTerminal(Terminal.kt:819)
        at com.stripe.stripeterminal.Terminal.initTerminal(Terminal.kt)
        at com.stripe.example.javaapp.MainActivity.initialize(MainActivity.java:390)
        at com.stripe.example.javaapp.MainActivity.requestPermissionsIfNecessarySdkBelow31(MainActivity.java:122)
        at com.stripe.example.javaapp.MainActivity.requestPermissionsIfNecessary(MainActivity.java:104)
        at com.stripe.example.javaapp.MainActivity.onCreate(MainActivity.java:75)
        at android.app.Activity.performCreate(Activity.java:6666)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2677)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2789)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1527)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:203)
        at android.app.ActivityThread.main(ActivityThread.java:6251)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "java.time.Duration" on path: DexPathList[[zip file "/data/app/com.stripe.example.javaapp-1/base.apk"],nativeLibraryDirectories=[/data/app/com.stripe.example.javaapp-1/lib/arm64, /data/app/com.stripe.example.javaapp-1/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /system/vendor/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.stripe.core.logging.dagger.MetricLoggerModuleKt.<clinit>(MetricLoggerModule.kt:26) 
        at com.stripe.core.logging.dagger.MetricLoggerModule.<init>(MetricLoggerModule.kt:31) 
        at com.stripe.core.logging.dagger.MetricLoggerModule.<init>(MetricLoggerModule.kt) 
        at com.stripe.stripeterminal.DaggerTerminalComponent$Builder.build(DaggerTerminalComponent.java:1679) 
        at com.stripe.stripeterminal.Terminal$Companion.initTerminal(Terminal.kt:860) 
        at com.stripe.stripeterminal.Terminal$Companion.initTerminal(Terminal.kt:819) 
        at com.stripe.stripeterminal.Terminal.initTerminal(Terminal.kt) 
        at com.stripe.example.javaapp.MainActivity.initialize(MainActivity.java:390) 
        at com.stripe.example.javaapp.MainActivity.requestPermissionsIfNecessarySdkBelow31(MainActivity.java:122) 
        at com.stripe.example.javaapp.MainActivity.requestPermissionsIfNecessary(MainActivity.java:104) 
        at com.stripe.example.javaapp.MainActivity.onCreate(MainActivity.java:75) 
        at android.app.Activity.performCreate(Activity.java:6666) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2677) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2789) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1527) 
        at android.os.Handler.dispatchMessage(Handler.java:110) 
        at android.os.Looper.loop(Looper.java:203) 
        at android.app.ActivityThread.main(ActivityThread.java:6251) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924) 
```

**Comment by billfinn-stripe** [comment]
Hi @minjie0707 and @anton8901 -- thanks for reporting and letting us know about this issue. The problem is that `java.time.Duration` became available in Android API level 26. The Java 8 compiler uses desugaring to make some of these core Java APIs available to lower Android API levels. However, desugaring ignores minification rules, so building a minified and desugared .apk that consumes the Terminal SDK will exhibit this error.

Can you look at your app's compile options and see if you are setting `coreLibraryDesugaringEnabled = true`? If so, can you remove that setting, recompile, and see if you are still able to reproduce this exception? Finally, can you confirm that you are only seeing these exceptions on devices that are running Android 7.1 or lower?

In the meantime, we are removing our usages of classes that were introduced in API levels greater than 21 (e.g., `java.time.Duration`), which will be our permanent fix for this problem.

**Comment by minjie0707** [comment]
> Hi @minjie0707 and @anton8901 -- thanks for reporting and letting us know about this issue. The problem is that `java.time.Duration` became available in Android API level 26. The Java 8 compiler uses desugaring to make some of these core Java APIs available to lower Android API levels. However, desugaring ignores minification rules, so building a minified and desugared .apk that consumes the Terminal SDK will exhibit this error.
> 
> Can you look at your app's compile options and see if you are setting `coreLibraryDesugaringEnabled = true`? If so, can you remove that setting, recompile, and see if you are still able to reproduce this exception? Finally, can you confirm that you are only seeing these exceptions on devices that are running Android 7.1 or lower?
> 
> In the meantime, we are removing our usages of classes that were introduced in API levels greater than 21 (e.g., `java.time.Duration`), which will be our permanent fix for this problem.

I do not set coreLibraryDesugaringEnabled = true . Yes , I  only  see the crash on  the devices that version is 7.1.2 , it is fine on android 11

**Comment by sjl-stripe** [comment]
@anton8901 @minjie0707 
We've just released [v2.10.2 of the Terminal Android SDK](https://github.com/stripe/stripe-terminal-android/releases/tag/v2.10.2) which contains the fix.

Closing this issue for now. Please re-open if the issue still happens on v2.10.2 or above.
